src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -1299,6 +1299,10 @@ protected void openConnection(final TaskListener listener, final SlaveComputer c
         int maxNumRetries = this.maxNumRetries == null || this.maxNumRetries < 0 ? 0 : this.maxNumRetries;
         for (int i = 0; i <= maxNumRetries; i++) {
             try {
+                // We pass launch timeout so that the connection will be able to abort once it reaches the timeout
+                // It is a poor man's logic, but it should cause termination if the connection goes strongly beyond the timeout
+                //TODO: JENKINS-48617 and JENKINS-48618 need to be implemented to make it fully robust
+                int launchTimeoutMillis = (int)getLaunchTimeoutMillis();
```

Comment:
Possible integer overflow? I suppose it is unlikely that we would want to wait 2³¹ms, but…

Reply:
Yeah, maybe I should add handling for that. But yeah, it it a bit paranoid in this case.
It's roughly 24.8 days 🤷‍♂️